pkg/gardenlet/controller/backupentry/backupentry/add.go

```diff
@@ -0,0 +1,145 @@
+	// the controller manually.
+	c, err := controller.New(
+		ControllerName,
+		mgr,
+		controller.Options{
+			Reconciler:              r,
+			MaxConcurrentReconciles: pointer.IntDeref(r.Config.ConcurrentSyncs, 0),
+			RecoverPanic:            true,
+			// if going into exponential backoff, wait at most the configured sync period
+			RateLimiter: r.RateLimiter,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := c.Watch(
+		source.NewKindWithCache(&gardencorev1beta1.BackupEntry{}, gardenCluster.GetCache()),
+		controllerutils.EnqueueCreateEventsOncePer24hDuration(r.Clock),
+		&predicate.GenerationChangedPredicate{},
+		r.SeedNamePredicate(),
+	); err != nil {
+		return err
+	}
+
+	return c.Watch(
+		source.NewKindWithCache(&extensionsv1alpha1.BackupEntry{}, seedCluster.GetCache()),
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapExtensionBackupEntryToCoreBackupEntry), mapper.UpdateWithNew, c.GetLogger()),
+		predicateutils.ExtensionStatusChanged(),
+	)
+}
+
+// SeedNamePredicate returns a predicate which returns true when the object belongs to this seed.
+func (r *Reconciler) SeedNamePredicate() predicate.Predicate {
+	return predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		backupEntry, ok := obj.(*gardencorev1beta1.BackupEntry)
+		if !ok {
+			return false
+		}
+		if backupEntry.Spec.SeedName == nil {
+			return false
+		}
+		if backupEntry.Status.SeedName == nil || *backupEntry.Spec.SeedName == *backupEntry.Status.SeedName {
+			return *backupEntry.Spec.SeedName == r.SeedName
+		}
+
+		return *backupEntry.Status.SeedName == r.SeedName
+	})
+}
+
+// MapExtensionBackupEntryToCoreBackupEntry is a mapper.MapFunc for mapping a extensions.gardener.cloud/v1alpha1.BackupEntry to the owning
+// core.gardener.cloud/v1beta1.BackupEntry.
+func (r *Reconciler) MapExtensionBackupEntryToCoreBackupEntry(ctx context.Context, log logr.Logger, _ client.Reader, obj client.Object) []reconcile.Request {
+	extensionBackupEntry, ok := obj.(*extensionsv1alpha1.BackupEntry)
+	if !ok {
+		return nil
+	}
+
+	shootTechnicalID, _ := gutil.ExtractShootDetailsFromBackupEntryName(extensionBackupEntry.Name)
+	if shootTechnicalID == "" {
+		return nil
+	}
+
+	namespaceName := getProjectNamespaceFromTechincalId(ctx, r.GardenClient, shootTechnicalID)
+	if namespaceName == "" {
+		return nil
+	}
+
+	return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: obj.GetName(), Namespace: namespaceName}}}
+}
+
+// TODO: Reuse function from https://github.com/gardener/gardener/pull/6839
+func getProjectNamespaceFromTechincalId(ctx context.Context, reader client.Reader, shootTechnicalID string) string {
+	var (
+		tokens      = strings.Split(shootTechnicalID, "--")
+		projectName = tokens[len(tokens)-2]
+
+		project = &gardencorev1beta1.Project{}
+	)
+
+	if err := reader.Get(ctx, kutil.Key(projectName), project); err != nil {
+		return ""
+	}
+	return *project.Spec.Namespace
+}
```

Comment:
Apart from the typo in the function name, the `ShootState` controller just uses the `Cluster` resource to get to this information (ref https://github.com/gardener/gardener/blob/master/pkg/extensions/cluster.go#L197-L227). Can we do the same?